src/System.Text.Json/src/System/Text/Json/ThrowHelper.cs

```diff
@@ -20,6 +20,142 @@ private static ArgumentException GetArgumentException(string message)
+            }
+            else
+            {
+                Debug.Assert(value.Length > JsonConstants.MaxCharacterTokenSize);
+                ThrowArgumentException(SR.Format(SR.ValueTooLarge, value.Length));
+            }
+        }
+
+        public static void ThrowArgumentException(ReadOnlySpan<char> propertyName, ReadOnlySpan<byte> value)
+        {
+            if (propertyName.Length > JsonConstants.MaxCharacterTokenSize)
+            {
+                ThrowArgumentException(SR.Format(SR.PropertyNameTooLarge, propertyName.Length));
+            }
+            else
+            {
+                Debug.Assert(value.Length > JsonConstants.MaxTokenSize);
+                ThrowArgumentException(SR.Format(SR.ValueTooLarge, value.Length));
+            }
+        }
+
+        public static void ThrowArgumentException(ReadOnlySpan<char> propertyName, ReadOnlySpan<char> value)
+        {
+            if (propertyName.Length > JsonConstants.MaxCharacterTokenSize)
+            {
+                ThrowArgumentException(SR.Format(SR.PropertyNameTooLarge, propertyName.Length));
+            }
+            else
+            {
+                Debug.Assert(value.Length > JsonConstants.MaxCharacterTokenSize);
+                ThrowArgumentException(SR.Format(SR.ValueTooLarge, value.Length));
+            }
+        }
+
+        public static void ThrowJsonWriterOrArgumentException(ReadOnlySpan<byte> propertyName, int currentDepth)
+        {
+            GetJsonWriterOrArgumentException(propertyName, currentDepth);
+        }
+
+        public static void ThrowJsonWriterException(string message)
+        {
+            throw GetJsonWriterException(message);
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        private static JsonWriterException GetJsonWriterException(string message)
+        {
+            return new JsonWriterException(message);
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        private static void GetJsonWriterOrArgumentException(ReadOnlySpan<byte> propertyName, int currentDepth)
+        {
+            currentDepth &= JsonConstants.RemoveFlagsBitMask;
+            if (currentDepth >= JsonConstants.MaxWriterDepth)
+            {
+                ThrowJsonWriterException(SR.Format(SR.DepthTooLarge, currentDepth));
+            }
+            else
+            {
+                Debug.Assert(propertyName.Length > JsonConstants.MaxCharacterTokenSize);
+                ThrowArgumentException(SR.Format(SR.PropertyNameTooLarge, propertyName.Length));
+            }
+        }
+
+        public static void ThrowJsonWriterOrArgumentException(ReadOnlySpan<char> propertyName, int currentDepth)
+        {
+            GetJsonWriterOrArgumentException(propertyName, currentDepth);
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        private static void GetJsonWriterOrArgumentException(ReadOnlySpan<char> propertyName, int currentDepth)
```

Comment:
Why is this called "Get" when it's not returning anything?  Same in some other cases.